### PR TITLE
Removing old custom RootFS behavior on Windows.

### PR DESCRIPTION
**Pull request by swernli**
@jhowardmsft @jstarks 
Windows base layers are no longer the special "layers+base" type, so we can remove all the special handling for that.

Signed-off-by: Stefan J. Wernli swernli@microsoft.com

### CODE REVIEW

**Comment by jhowardmsft** [comment]
Nice! LGTM once Janky passes. Has some import fixups required:

```
14:10:12 distribution/pull_v2_unix.go:8: imported and not used: "github.com/docker/distribution/manifest/schema1"
14:10:12 distribution/pull_v2_unix.go:9: imported and not used: "github.com/docker/docker/image"
```

**Comment by jstarks** [comment]
LGtm

**Comment by thaJeztah** [comment]
LGTM, but

ping @aaronlehmann @stevvooe for the change in image-specs

File: daemon/oci_windows.go [hunk 1]
```diff
@@ -89,15 +89,10 @@ func (daemon *Daemon) createSpec(c *container.Container) (*libcontainerd.Spec, e
 
 	// s.Windows.LayerPaths
 	var layerPaths []string
-	if img.RootFS != nil && (img.RootFS.Type == image.TypeLayers || img.RootFS.Type == image.TypeLayersWithBase) {
+	if img.RootFS != nil && (img.RootFS.Type == image.TypeLayers) {
```
Comment (aaronlehmann) [inline daemon/oci_windows.go]: nit: extra parentheses

**Comment by aaronlehmann** [comment]
Seems fine to me. The base layer stuff was hack that we didn't want to support long-term. It's great that we're able to remove it.

Normally there would be backward compatibility concerns, but I know the Windows daemon is not in wide use and there are not many images for it, so if you feel it's okay to get rid of this as a step change, that's fine with me.

Needs a rebase though.

cc @dmcgowan @tonistiigi

File: image/spec/v1.1.md [hunk 2]
```diff
@@ -370,10 +370,7 @@ whitespace. It has been added to this example for clarity.
 
         <ul>
           <li>
-            <code>type</code> is usually set to <code>layers</code>. There is
-            also a Windows-specific value <code>layers+base</code> that allows
-            a base layer to be specified in a field of <code>rootfs</code>
-            called <code>base_layer</code>.
+            <code>type</code> is usually set to <code>layers</code>.
```
Comment (stevvooe) [inline image/spec/v1.1.md]: We should probably leave this in or [omit from OCI](https://github.com/opencontainers/image-spec/blob/master/serialization.md#image-json-field-descriptions) if we don't intend to support it going forward.

Comment (swernli) [inline image/spec/v1.1.md]: Good point.  I'll file an issue against @jstarks to get this removed from OCI.

Comment (swernli) [inline image/spec/v1.1.md]: Actually, looks like you already opened that issue!

**Comment by LK4D4** [comment]
I think it has plenty of lgtms

**Pull request merged**